src/utils/hooks/useLoadApplicationNew.ts

```diff
@@ -0,0 +1,201 @@
+  TemplateElementState,
+  UseGetApplicationProps,
+  User,
+} from '../types'
+import evaluateExpression from '@openmsupply/expression-evaluator'
+import useTriggerProcessing from './useTriggerProcessing'
+import { getApplicationSections } from '../helpers/application/getSectionsDetails'
+import { buildSectionsStructure } from '../helpers/structure/buildSectionsStructureNEW'
+import {
+  Application,
+  ApplicationSection,
+  Template,
+  TemplateElement,
+  TemplateStage,
+  useGetApplicationNewQuery,
+  useGetApplicationStatusQuery,
+} from '../generated/graphql'
+
+const useLoadApplication = ({
+  currentUser,
+  serialNumber,
+  networkFetch,
+}: UseGetApplicationProps) => {
+  const [isLoading, setIsLoading] = useState(true)
+  const [structureError, setStrucutreError] = useState('')
+  const [structure, setFullStructure] = useState<FullStructure>()
+  const [template, setTemplate] = useState<TemplateDetails>()
+
+  const { data, error } = useGetApplicationNewQuery({
+    variables: {
+      serial: serialNumber,
+    },
+    // pollInterval: 500,
+    // skip: isLoading,
+    fetchPolicy: networkFetch ? 'network-only' : 'cache-first',
+  })
+
+  // const { error: triggerError, isTriggerProcessing } = useTriggerProcessing({
+  //   isApplicationLoaded: data !== undefined,
+  //   serialNumber,
+  //   // triggerType: 'applicationTrigger',
+  // })
+
+  // const { data: statusData, error: statusError } = useGetApplicationStatusQuery({
+  //   variables: { serial: serialNumber },
+  //   skip: isTriggerProcessing,
+  //   fetchPolicy: 'network-only',
+  // })
+
+  useEffect(() => {
+    // if (!data || !statusData || isTriggerProcessing) return
+    if (!data) return
+    const application = data.applicationBySerial as Application
+
+    if (!application) {
+      setStrucutreError('No application found')
+    }
+
+    const { id, code, name, startMessage } = application.template as Template
+
+    setTemplate({
+      id,
+      code,
+      name: name as string,
+      startMessage: startMessage ? startMessage : undefined,
+    })
+
+    const applicationSection = application.applicationSections.nodes as ApplicationSection[]
+    const sections = getApplicationSections(applicationSection)
+
+    const templateStages = application.template?.templateStages.nodes as TemplateStage[]
+
+    const stagesDetails: ApplicationStages = {
+      stages: templateStages.map((stage) => ({
+        number: stage.number as number,
+        title: stage.title as string,
+        description: stage.description ? stage.description : undefined,
+      })),
+      submissionMessage: application.template?.submissionMessage as string,
+    }
+
+    const applicationDetails: ApplicationDetails = {
+      id: application.id,
+      type: application.template?.name as string,
+      isLinear: application.template?.isLinear as boolean,
+      serial: application.serial as string,
+      name: application.name as string,
+      outcome: application.outcome as string,
```

Comment:
Yes, that's probably why, since the `id` doesn't need that and can't be null on the database. I'll add a back-end issue associated with this change.

Reply:
Great, thanks -- can you note the file/line of this in the issue notes so we remember to fix this at the same time? Cheers.

Reply:
Changing to use NOT NULL on fields didn't solve it... I tried changing on the `codegen.yml` to use the `avoidOptipnals` flag as described in the [docs](https://graphql-code-generator.com/docs/plugins/typescript)  which does:
`This will cause the generator to avoid using TypeScript optionals (?) on types, so the following definition: type A { myField: String } will output myField: Maybe<string> instead of myField?: Maybe<string>.`

But that ended up resolving this issue, since it's a **Maybe** value it still needs casting. So I'll leave this as it is, but now we know we can remove those 3 possible types `Object | null | undefined` from all fields returned if we just use this flag - warning: it will cause many errors, the reason why I didn't want to make this change now.